File: Arrays/Python/sort_array_without_builtin.py
arr = [5, 1, 4, 2, 8]

for i in range(len(arr)):
    swapped = False
    for j in range(0, len(arr)-i-1):
        if arr[j] > arr[j+1]:
            arr[j], arr[j+1] = arr[j+1], arr[j]
            swapped = True
    if not swapped:
        break
    
print(arr)

File: Arrays/Java/SortArrayWithoutBuiltin.java
import java.util.Arrays;

public class SortArrayWithoutBuiltin {
    public static void main(String[] args) {
        int[] arr = {5, 1, 4, 2, 8};

        for (int i = 0; i < arr.length - 1; i++) {
            boolean swapped = false;
            for (int j = 0; j < arr.length - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                    swapped = true;
                }
            }
            if (!swapped) break; // Already sorted → stop early
        }

        System.out.println(Arrays.toString(arr));
    }
}